p2p/chain.py

```diff
@@ -543,28 +561,10 @@ class RegularChainSyncer(FastChainSyncer):
     """
     _exit_on_sync_complete = False
 
-    async def _handle_msg(self, peer: HeaderRequestingPeer, cmd: protocol.Command,
-                          msg: protocol._DecodedMsgType) -> None:
-        peer = cast(ETHPeer, peer)
-        if isinstance(cmd, eth.BlockHeaders):
-            self._handle_block_headers(tuple(cast(Tuple[BlockHeader, ...], msg)))
-        elif isinstance(cmd, eth.BlockBodies):
-            await self._handle_block_bodies(peer, list(cast(Tuple[eth.BlockBody], msg)))
-        elif isinstance(cmd, eth.NewBlock):
-            await self._handle_new_block(peer, cast(Dict[str, Any], msg))
-        elif isinstance(cmd, eth.GetBlockHeaders):
-            await self._handle_get_block_headers(peer, cast(Dict[str, Any], msg))
-        elif isinstance(cmd, eth.GetBlockBodies):
-            await handle_get_block_bodies(
-                self.db, peer, cast(List[Hash32], msg), self.logger, self.cancel_token)
-        elif isinstance(cmd, eth.GetReceipts):
-            await handle_get_receipts(
-                self.db, peer, cast(List[Hash32], msg), self.logger, self.cancel_token)
-        elif isinstance(cmd, eth.GetNodeData):
-            await handle_get_node_data(
-                self.db, peer, cast(List[Hash32], msg), self.logger, self.cancel_token)
-        else:
-            self.logger.debug("%s msg not handled yet, need to be implemented", cmd)
+    async def _handle_block_receipts(
+            self, peer: ETHPeer, receipts_by_block: List[List[eth.Receipt]]) -> None:
+        # When doing a regular sync we never request receipts.
+        self.logger.warn("Unexpected BlockReceipts msg from %s", peer)
```

Comment:
Any reason this was not inlined in the `if/elif` block on lines 485-500 the same way that the `NodeData` warning is inlined.  Similarly, should we disconnect in this case?

Reply:
lines 485-500 are in `FastChainSyncer`, and during a fast sync we send `GetReceipts` requests and expect `Receipts` responses. This is `RegularChainSyncer`, and during a regular (i.e. block-processing) sync we don't send `GetReceipts`. I guess it'd make sense to disconnect as well, yeah

Reply:
`NodeData`, on the other hand, is unexpected in both syncers -- it is only the `StateDownloader` that requests those

Reply:
I mis-read the diff and didn't see that those were two different classes.